bitnami/thanos/templates/storegateway/service-headless.yaml

```diff
@@ -1,22 +0,0 @@
-{{- if .Values.storegateway.enabled -}}
-apiVersion: v1
-kind: Service
-metadata:
-  name: {{ include "thanos.fullname" . }}-storegateway-headless
-  labels: {{- include "thanos.labels" . | nindent 4 }}
-    app.kubernetes.io/component: storegateway
-spec:
-  type: ClusterIP
-  clusterIP: None
-  ports:
-    - port: {{ .Values.storegateway.service.http.port }}
-      targetPort: http
-      protocol: TCP
-      name: http
-    - port: {{ .Values.storegateway.service.grpc.port }}
-      targetPort: grpc
-      protocol: TCP
-      name: rcp
-  selector: {{- include "thanos.matchLabels" . | nindent 4 }}
-    app.kubernetes.io/component: storegateway
-{{- end }}
```

Comment:
Why did you remove this headless service? It's useful so I can have bot a headless service and a non-headless service for StoreGateway.

Reply:
It seems to me that a dedicated headless service is redundant.
For both described cases, we can use one headless service.
Anyway, Thanos Querier gets a list of endpoints from the service.

Reply:
> It seems to me that a dedicated headless service is redundant.

It's useful when you want to provide both the features below simultaneously:

- A) Provide a "single" access point for users, so they don't need to worry about the specific replica they're accessing. They just access some service hostname/IP and forget about the rest.
- B) Provide a stable network identifier to every replicas so the different micro-services can target a specific replica when needed.

I don't know under what circumstances the behaviour above is desired on Thanos, but keeping these dedicated headless services won't make any harm, and we'll cover that possibility for those users that find it useful.

Reply:
Yes, you're right - I returned headless service.
I just added new parameter "storegateway.service.additionalHeadless" and "ruler.service.additionalHeadless" to enable/disable this service.
+ adjusted servicemonitor to avoid duplicated metrics.